FTBFS on Arch Linux x86_64

I tried building core_extended.114.35+03 from Jane Street's OPAM repository. This fails when building some C code:

```
Running[9]: (cd _build/default/src && /home/marek/.opam/4.04.0/bin/ocamlc.opt -g -I /home/marek/.opam/4.04.0/lib/base -I /home/marek/.opam/4.04.0/lib/core -I /home/marek/.opam/4.04.0/lib/core_kernel -I /home/marek/.opam/4.04.0/lib/jane-street-headers -ccopt -g -o extended_unix_stubs.o extended_unix_stubs.c)

Command [9] exited with code 2:
$ (cd _build/default/src && /home/marek/.opam/4.04.0/bin/ocamlc.opt -g -I /home/marek/.opam/4.04.0/lib/base -I /home/marek/.opam/4.04.0/lib/core -I /home/marek/.opam/4.04.0/lib/core_kernel -I /home/marek/.opam/4.04.0/lib/jane-street-headers -ccopt -g -o extended_unix_stubs.o extended_unix_stubs.c)
extended_unix_stubs.c: In function ‘quota_query’:
extended_unix_stubs.c:107:15: warning: passing argument 1 of ‘quotactl’ makes integer from pointer without a cast [-Wint-conversion]
      quotactl((device), (cmd), (id), (parg))
               ^
extended_unix_stubs.c:156:7: note: in expansion of macro ‘quota_control’
   if (quota_control(String_val(v_path), cmd, id, (caddr_t)&quota))
       ^~~~~~~~~~~~~
In file included from extended_unix_stubs.c:19:0:
/usr/include/sys/quota.h:130:12: note: expected ‘int’ but argument is of type ‘char *’
 extern int quotactl (int __cmd, const char *__special, int __id,
            ^~~~~~~~
extended_unix_stubs.c:107:25: warning: passing argument 2 of ‘quotactl’ makes pointer from integer without a cast [-Wint-conversion]
      quotactl((device), (cmd), (id), (parg))
                         ^
extended_unix_stubs.c:156:7: note: in expansion of macro ‘quota_control’
   if (quota_control(String_val(v_path), cmd, id, (caddr_t)&quota))
       ^~~~~~~~~~~~~
In file included from extended_unix_stubs.c:19:0:
/usr/include/sys/quota.h:130:12: note: expected ‘const char *’ but argument is of type ‘int’
 extern int quotactl (int __cmd, const char *__special, int __id,
            ^~~~~~~~
extended_unix_stubs.c:109:43: error: ‘struct dqblk’ has no member named ‘dqb_curbytes’; did you mean ‘dqb_curinodes’?
 #  define QUOTA_SPACE_USED(quota) ((quota).dqb_curbytes)
                                           ^
extended_unix_stubs.c:159:55: note: in expansion of macro ‘QUOTA_SPACE_USED’
   bytes_used = QUOTA_BYTES_PER_SPACE_UNIT * (int64_t) QUOTA_SPACE_USED(quota);
                                                       ^~~~~~~~~~~~~~~~
extended_unix_stubs.c: In function ‘quota_modify’:
extended_unix_stubs.c:107:15: warning: passing argument 1 of ‘quotactl’ makes integer from pointer without a cast [-Wint-conversion]
      quotactl((device), (cmd), (id), (parg))
               ^
extended_unix_stubs.c:204:7: note: in expansion of macro ‘quota_control’
   if (quota_control(String_val(v_path), cmd, id, (caddr_t)&quota))
       ^~~~~~~~~~~~~
In file included from extended_unix_stubs.c:19:0:
/usr/include/sys/quota.h:130:12: note: expected ‘int’ but argument is of type ‘char *’
 extern int quotactl (int __cmd, const char *__special, int __id,
            ^~~~~~~~
extended_unix_stubs.c:107:25: warning: passing argument 2 of ‘quotactl’ makes pointer from integer without a cast [-Wint-conversion]
      quotactl((device), (cmd), (id), (parg))
                         ^
extended_unix_stubs.c:204:7: note: in expansion of macro ‘quota_control’
   if (quota_control(String_val(v_path), cmd, id, (caddr_t)&quota))
       ^~~~~~~~~~~~~
In file included from extended_unix_stubs.c:19:0:
/usr/include/sys/quota.h:130:12: note: expected ‘const char *’ but argument is of type ‘int’
 extern int quotactl (int __cmd, const char *__special, int __id,
            ^~~~~~~~
```

--- a/src/extended_unix_stubs.c
+++ b/src/extended_unix_stubs.c
@@ -101,7 +101,7 @@ CAMLprim value getloadavg_stub (value v_unit __unused)
 #  define QUOTA_MODIFY_COMMAND Q_SETQUOTA
 #  define QUOTA_SET_VALID_FIELDS(quota) ((void)quota)
 
-#elif !(defined _LINUX_QUOTA_VERSION) /* Mac OS */
+#elif defined (__APPLE__) /* Mac OS */
 
 #  define quota_control(device, cmd, id, parg)  \
      quotactl((device), (cmd), (id), (parg))
@@ -110,7 +110,7 @@ CAMLprim value getloadavg_stub (value v_unit __unused)
 #  define QUOTA_MODIFY_COMMAND Q_SETQUOTA
 #  define QUOTA_SET_VALID_FIELDS(quota) ((void)quota)
 
-#elif _LINUX_QUOTA_VERSION < 2
+#elif defined (_LINUX_QUOTA_VERSION) && _LINUX_QUOTA_VERSION < 2
 
 #  define quota_control(device, cmd, id, parg)  \
      quotactl((cmd), (device), (id), (parg))
@@ -119,7 +119,7 @@ CAMLprim value getloadavg_stub (value v_unit __unused)
 #  define QUOTA_MODIFY_COMMAND Q_SETQLIM
 #  define QUOTA_SET_VALID_FIELDS(quota) ((void)quota)
 
-#else /* _LINUX_QUOTA_VERSION >= 2 */
+#else /* _LINUX_QUOTA_VERSION >= 2 or not defined, GLIBC 2.25+ */
 
 #  define quota_control(device, cmd, id, parg)  \
      quotactl((cmd), (device), (id), (parg))

